importer_client/python/timesketch_import_client/importer.py

```diff
@@ -708,21 +708,24 @@ def celery_task_id(self):
         """Return the celery task identification for the upload."""
         return self._celery_task_id
 
-    def close(self):
-        """Close the streamer."""
+    def close(self, analyzer_names=None):
+        """Close the streamer, and run the analyzers based on the analyzer_names"""
         try:
             self._ready()
         except ValueError:
             return
 
+        if analyzer_names is None:
+            analyzer_names = []
+
         if self._data_lines:
             self.flush(end_stream=True)
 
         # Trigger auto analyzer pipeline to kick in.
         pipe_resource = "{0:s}/sketches/{1:d}/analyzer/".format(
             self._sketch.api.api_root, self._sketch.id
         )
-        data = {"index_name": self._index}
+        data = {"index_name": self._index, "analyzer_names": analyzer_names}
         _ = self._sketch.api.session.post(pipe_resource, json=data)
```

Comment:
You only need to execute this part if the user requested any analyzers to run. I would recommend to do a check for provided analyzer_names.